scripts/check_if_pr_is_low_risk.py

```diff
@@ -0,0 +1,317 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Check whether a PR is low-risk.
+
+A low-risk PR can be merged without running the full CI checks. When
+called with a URL to a PR, this script exits with code 0 if and only if
+the PR is low-risk. Otherwise, the exit code is nonzero.
+"""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import argparse
+import json
+import re
+import sys
+
+import python_utils
+from scripts import common
+
+
+GITHUB_API_PR_ENDPOINT = (
+    'https://api.github.com/repos/%s/%s/pulls/%s')
+PR_URL_REGEX = (
+    r'^https://github.com/(?P<owner>\w+)/(?P<repo>\w+)/pull/(?P<num>\w+)/?$')
+UPSTREAM_REMOTE = 'upstream'
+FILES_THAT_NEED_DIFFS = (
+    'package.json'
+    'core/templates/pages/about-page/about-page.constants.ts',
+)
+
+
+def parse_pr_url(pr_url):
+    """Extract the owner, repo, and PR number from a PR URL.
+
+    For example, in the URL https://github.com/foobar/oppia/pull/23, the
+    owner is `foobar`, the repo is `oppia`, and the PR number is `23`.
+
+    Args:
+        pr_url: str. URL to the PR on GitHub.
+
+    Returns:
+        str, str, str. 3-tuple of the PR owner, repository, and number.
+    """
+    match = re.match(PR_URL_REGEX, pr_url)
+    if match:
+        return match.group('owner', 'repo', 'num')
+    else:
+        return None
+
+
+def load_diff(base_branch):
+    """Load the diff between the head and base.
+
+    Only determine the diffs for files in FILES_THAT_NEED_DIFFS. Other
+    files will be listed as having changed, but their diff lines will
+    not be returned.
+
+    Args:
+        base_branch: str. Base branch of PR.
+
+    Returns:
+        tuple(list(tuple(str, str)), dict(str, list(str)). Tuple of a
+        list of changed files (each a tuple of before, after) and a
+        dictionary mapping from file names to list of diff lines for
+        each file. In the event of a parsing error, a tuple of an empty
+        list and empty dictionary is returned.
+    """
+    diff_name_status = common.run_cmd([
+        'git', 'diff', '--name-status',
+        '{}/{}'.format(UPSTREAM_REMOTE, base_branch),
+    ])
+    diff_files = []
+    for line in diff_name_status.split('\n'):
+        if not line:
+            continue
+        split = line.split()
+        if len(split) < 2 or len(split) > 3:
+            python_utils.PRINT(
+                'Failed to parse diff --name-status line "%s"'
+                % line)
+            return [], {}
+        elif len(split) == 2:
+            diff_files.append((split[1], split[1]))
+        elif len(split) == 3:
+            diff_files.append((split[1], split[2]))
+    file_diffs = {}
+    for file_tuple in diff_files:
+        for filename in file_tuple:
+            if filename in file_diffs:
+                continue
```

Comment:
This avoid re-generating diffs. For example, if a file modified without a name change, it will appear twice in `file_tuple`--once for the "before" name and once for the "after" name. This check avoids calling `git diff` on the file a second time